src/hnsw/validate_index.c

```diff
@@ -0,0 +1,650 @@
+#include <postgres.h>
+
+#include "hnsw/validate_index.h"
+
+#include <access/heapam.h>  /* relation_open */
+#include <assert.h>         /* assert */
```

Comment:
Can we use ldb_invariant in all places where assert is used?
Assertion failures do not propagate to the client, so during development it is harder to see the output of an `assert` failure than an `ldb_invariant` failure